plugin/trino-querylog/src/main/java/io/trino/plugin/querylog/QuerylogListener.java

```diff
@@ -0,0 +1,161 @@
+
+    private final OkHttpClient client;
+
+    /**
+     * Create a HTTP logger of query events
+     * @param inetAddress Hostname to destination
+     * @param port Port
+     * @param httpHeaders Map of custom HTTP headers
+     * @param shouldLogCreated Enable logging of {@link io.trino.spi.eventlistener.QueryCreatedEvent}
+     * @param shouldLogCompleted Enable logging of {@link io.trino.spi.eventlistener.QueryCompletedEvent}
+     * @param shouldLogSplit Enable logging of {@link io.trino.spi.eventlistener.SplitCompletedEvent}
+     */
+    public QuerylogListener(InetAddress inetAddress, int port, Map<String, String> httpHeaders, boolean shouldLogCreated, boolean shouldLogCompleted, boolean shouldLogSplit)
+    {
+        this.shouldLogCreated = shouldLogCreated;
+        this.shouldLogCompleted = shouldLogCompleted;
+        this.shouldLogSplit = shouldLogSplit;
+        this.httpHeaders = httpHeaders;
+        this.inetAddress = inetAddress;
+        this.scheme = "http";
+        this.port = port;
+
+        client = new OkHttpClient();
+    }
+
+    /**
+     * Create a HTTPS logger of query events
+     * @param inetAddress Hostname to destination
+     * @param port Port
+     * @param socketFactory SSLSocketFactory with appropriate certificates
+     * @param trustManager X509TrustManager
+     * @param httpHeaders Map of custom HTTP headers
+     * @param shouldLogCreated Enable logging of {@link io.trino.spi.eventlistener.QueryCreatedEvent}
+     * @param shouldLogCompleted Enable logging of {@link io.trino.spi.eventlistener.QueryCompletedEvent}
+     * @param shouldLogSplit Enable logging of {@link io.trino.spi.eventlistener.SplitCompletedEvent}
+     */
+    public QuerylogListener(InetAddress inetAddress, int port, SSLSocketFactory socketFactory, X509TrustManager trustManager, Map<String, String> httpHeaders, boolean shouldLogCreated, boolean shouldLogCompleted, boolean shouldLogSplit)
+    {
+        this.shouldLogCreated = shouldLogCreated;
+        this.shouldLogCompleted = shouldLogCompleted;
+        this.shouldLogSplit = shouldLogSplit;
+        this.httpHeaders = httpHeaders;
+        this.inetAddress = inetAddress;
+        this.scheme = "https";
+        this.port = port;
+
+        this.client = new OkHttpClient.Builder().sslSocketFactory(socketFactory, trustManager).build();
+    }
+
+    @Override
+    public void queryCreated(QueryCreatedEvent queryCreatedEvent)
+    {
+        if (shouldLogCreated) {
+            sendLog(queryCreatedEvent);
+        }
+    }
+
+    @Override
+    public void queryCompleted(QueryCompletedEvent queryCompletedEvent)
+    {
+        if (shouldLogCompleted) {
+            sendLog(queryCompletedEvent);
+        }
+    }
+
+    @Override
+    public void splitCompleted(SplitCompletedEvent splitCompletedEvent)
+    {
+        if (shouldLogSplit) {
+            sendLog(splitCompletedEvent);
+        }
+    }
+
+    private <T> void sendLog(T event)
```

Comment:
Doing logging synchronously here does not feel right to me. I would suggest to use a separate executor for doing HTTP requests here.

Reply:
I'm curious about something - are these event listeners tied in synchronously to the query execution? If so, it makes sense to make this run in a separate executor, but I don't see the value otherwise.

Reply:
IIRC they are often called out directly from the threads which are responsible for other parts of query execution. And spending too much time in listener may result in blocking Trino operation.

Reply:
yes, in that case it makes perfect sense to do this in a separate thread. @mosiac1 can we please try to add this?